### PR TITLE
Fix #2752

### LINKED ISSUE #2752: Editing set-cookies while no response is present crashes mitmproxy
##### Steps to reproduce the problem:

1. Create a new flow by pressing `n`.
2. Enter, e -> set-cookies

```
Traceback (most recent call last):
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/master.py", line 216, in run
    self.loop.run()
  File "/home/user/venv/lib/python3.5/site-packages/urwid/main_loop.py", line 278, in run
    self._run()
  File "/home/user/venv/lib/python3.5/site-packages/urwid/main_loop.py", line 376, in _run
    self.event_loop.run()
  File "/home/user/venv/lib/python3.5/site-packages/urwid/main_loop.py", line 682, in run
    self._loop()
  File "/home/user/venv/lib/python3.5/site-packages/urwid/main_loop.py", line 719, in _loop
    self._watch_files[fd]()
  File "/home/user/venv/lib/python3.5/site-packages/urwid/raw_display.py", line 393, in <lambda>
    event_loop, callback, self.get_available_raw_input())
  File "/home/user/venv/lib/python3.5/site-packages/urwid/raw_display.py", line 493, in parse_input
    callback(processed, processed_codes)
  File "/home/user/venv/lib/python3.5/site-packages/urwid/main_loop.py", line 403, in _update
    self.process_input(keys)
  File "/home/user/venv/lib/python3.5/site-packages/urwid/main_loop.py", line 503, in process_input
    k = self._topmost_widget.keypress(self.screen_size, k)
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/window.py", line 299, in keypress
    k = super().keypress(size, k)
  File "/home/user/venv/lib/python3.5/site-packages/urwid/container.py", line 1128, in keypress
    return self.body.keypress( (maxcol, remaining), key )
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/window.py", line 36, in keypress
    ret = super().keypress(size, key)
  File "/home/user/venv/lib/python3.5/site-packages/urwid/container.py", line 1128, in keypress
    return self.body.keypress( (maxcol, remaining), key )
  File "/home/user/venv/lib/python3.5/site-packages/urwid/container.py", line 592, in keypress
    *self.calculate_padding_filler(size, True)), key)
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/overlay.py", line 130, in keypress
    self.master.keymap.handle("global", key)
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/keymap.py", line 130, in handle
    return self.executor(b.command)
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/commandexecutor.py", line 17, in __call__
    ret = self.master.commands.call(cmd)
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/command.py", line 221, in call
    return self.call_args(parts[0], parts[1:])
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/command.py", line 212, in call_args
    return self.commands[path].call(args)
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/command.py", line 101, in call
    ret = self.func(*pargs)
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/command.py", line 251, in wrapper
    return function(*args, **kwargs)
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/consoleaddons.py", line 174, in nav_select
    self.master.inject_key("m_select")
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/master.py", line 181, in inject_key
    self.loop.process_input([key])
  File "/home/user/venv/lib/python3.5/site-packages/urwid/main_loop.py", line 503, in process_input
    k = self._topmost_widget.keypress(self.screen_size, k)
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/window.py", line 299, in keypress
    k = super().keypress(size, k)
  File "/home/user/venv/lib/python3.5/site-packages/urwid/container.py", line 1128, in keypress
    return self.body.keypress( (maxcol, remaining), key )
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/window.py", line 36, in keypress
    ret = super().keypress(size, key)
  File "/home/user/venv/lib/python3.5/site-packages/urwid/container.py", line 1128, in keypress
    return self.body.keypress( (maxcol, remaining), key )
  File "/home/user/venv/lib/python3.5/site-packages/urwid/container.py", line 592, in keypress
    *self.calculate_padding_filler(size, True)), key)
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/overlay.py", line 120, in keypress
    self.callback(self.choices[self.walker.index])
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/consoleaddons.py", line 264, in callback
    self.master.commands.call(subcmd + " " + repl)
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/command.py", line 221, in call
    return self.call_args(parts[0], parts[1:])
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/command.py", line 212, in call_args
    return self.commands[path].call(args)
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/command.py", line 101, in call
    ret = self.func(*pargs)
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/command.py", line 251, in wrapper
    return function(*args, **kwargs)
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/consoleaddons.py", line 409, in edit_focus
    self.master.switch_view("edit_focus_setcookies")
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/master.py", line 236, in switch_view
    self.window.push(name)
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/window.py", line 223, in push
    self.focus_changed()
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/window.py", line 200, in focus_changed
    i.call("focus_changed")
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/window.py", line 110, in call
    getattr(self.top_window(), name)(*args, **kwargs)
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/grideditor/base.py", line 471, in focus_changed
    self.get_data(self.master.view.focus.flow),
  File "/mnt/c/Users/user/git/mitmproxy/mitmproxy/tools/console/grideditor/editors.py", line 149, in get_data
    return self.data_in(flow.response.cookies.items(multi=True))
AttributeError: 'NoneType' object has no attribute 'cookies'
```

##### System information

Mitmproxy: 3.0.0.dev0002 (commit 299df48)
Python: 3.5.2
OpenSSL: OpenSSL 1.1.0g 2 Nov 2017
Platform: Linux-4.4.0-43-Microsoft-x86_64-with-Ubuntu-16.04-xenial

### PR DESCRIPTION
**Pull request by kajojify**
There were 2 variants of how to fix it:
1) Raise `CommandError`, if flow hasn't response. So, when we try to edit response-related part of flow, mitmproxy will display `Flow has no response` in status bar. But it contradicts the text in empty response tab - `Press e and edit any aspect to add one` and the behaviour of 2.0.X version.
2) Make something like dummy response.

I chose second variant according to 2.0.X version
(https://github.com/mitmproxy/mitmproxy/blob/2.0.x/mitmproxy/tools/console/flowview.py#L368-L372). 
When we choose `response-headers`, `response-body` or `set-cookies` to edit, an empty response(https://github.com/mitmproxy/mitmproxy/blob/master/mitmproxy/net/http/response.py#L69) is added into the flow.

### CODE REVIEW

File: mitmproxy/tools/console/consoleaddons.py [hunk 1]
```diff
@@ -378,6 +379,8 @@ def edit_focus(self, part: str) -> None:
         # but for now it is.
         if not flow:
             raise exceptions.CommandError("No flow selected.")
+        if part in ("response-headers", "response-body", "set-cookies"):
+            flow.response = http.HTTPResponse.make()
```
Comment (mhils) [inline mitmproxy/tools/console/consoleaddons.py]: Pretty bad idea to always overwrite the response 😉

Comment (kajojify) [reply mitmproxy/tools/console/consoleaddons.py]: @mhils sorry, bad idea, lost it. Give me a sec to change it.

File: mitmproxy/tools/console/consoleaddons.py [hunk 2]
```diff
@@ -378,6 +379,9 @@ def edit_focus(self, part: str) -> None:
         # but for now it is.
         if not flow:
             raise exceptions.CommandError("No flow selected.")
+        if part in ("response-headers", "response-body", "set-cookies"):
+            if flow.response is None:
+                flow.response = http.HTTPResponse.make()
```
Comment (Kriechi) [inline mitmproxy/tools/console/consoleaddons.py]: To make the code a bit more precise and clearer to read: please combine the two conditions into a single if.

Comment (kajojify) [reply mitmproxy/tools/console/consoleaddons.py]: @Kriechi I thought about this, but one line will be more, than 79 symbols. Is it allowed to break this rule in Mitmproxy sometimes?

Comment (Kriechi) [reply mitmproxy/tools/console/consoleaddons.py]: hehe - true 😄 
You can add a line break with `\`, or extract all condition into a bool variable, similar to https://github.com/mitmproxy/mitmproxy/blob/master/mitmproxy/proxy/root_context.py#L109-L114.

**Pull request merged**